### PR TITLE
feat: add serviceinfo api specification draft

### PR DESCRIPTION
**Pull request by puiterwijk**
This adds an initial version of the serviceinfo api specification.

Signed-off-by: Patrick Uiterwijk <patrick@puiterwijk.org>

### CODE REVIEW

File: specs/serviceinfo_api.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+### Authentication
+
+For this protocol, the FDO Owner Onboarding Server will authenticate to the server of this API, via any of the authentication methods defined in the [OV Management API specification](https://github.com/fedora-iot/fido-device-onboard-rs/blob/main/specs/ov_management_api.md#authentication).
+
+## Endpoint
+
+This API consists of a single endpoint, the URL of which is configurable.
+The URL will have configurable fields for the device GUID and the list of supported modules.
+The modules are comma-seperated.
```
Comment (jnogol) [inline specs/serviceinfo_api.md:27]: Little typo: comma-sep**a**rated

File: specs/serviceinfo_api.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+The following keys are defined for the response to this endpoint:
+
+- `com.redhat.subscription_identity_certificate`: a string containing a PEM-encoded identity certificate for the system.
+- `initial_user`: a JSON object containing information about an initial user to be configured. Supported sub-keys are:
+    - `username`: the username of the user to configure.
+    - `ssh_keys`: a list of strings containing SSH keys to configure for this user.
+
+### Examples
+
+This assumes the URL is configured as `/device_info?device_guid={{device_guid}}&modules=&{{modules}}`.
```
Comment (jnogol) [inline specs/serviceinfo_api.md:47]: Would it make sense to version this API?

File: specs/serviceinfo_api.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+```
+
+#### Successful response
+
+``` HTTP
+HTTP/1.1 200 OK
+Content-Type: application/json
+Server: FDO-ServiceInfo-Server/1.0
+
+{"com.redhat.subscription_identity_certificate": "-----BEGIN CERTIFICATE-----\n......\n-----END CERTIFICATE-----", "initial_user": {"username": "root", "ssh_keys": ["ssh-rsa ...."]}, "undefined": "something_ignored"}
```
Comment (jnogol) [inline specs/serviceinfo_api.md:66]: In this example we're including `undefined` key, but this key is not explained in `Supported response keys`, should we include it?

Comment (puiterwijk) [reply specs/serviceinfo_api.md:66]: No, this is explicitly intended to show that the server can return fields that aren't defined.

File: specs/serviceinfo_api.md (hunk 2 above)
Comment (puiterwijk) [reply specs/serviceinfo_api.md]: Yep, done.

File: specs/serviceinfo_api.md (hunk 3 above)
Comment (jnogol) [reply specs/serviceinfo_api.md:67]: Makes sense.

**Review by jnogol — APPROVED** [review]

**Pull request merged**